katz/src/main/kotlin/katz/effects/internal/AndThen.kt

```diff
@@ -0,0 +1,131 @@
+package katz.effects.internal
+
+/**
+ * Abandon all hope
```

Comment:
Could you add some docs here about the purpose of this class ? I am trying to learn something here and that could be helpful :D